src/client/Client.cc

```diff
@@ -7659,6 +7720,9 @@ int Client::_write(Fh *f, int64_t offset, uint64_t size, const char *buf)
   }
 
   put_cap_ref(in, CEPH_CAP_FILE_WR);
+  if (buf == NULL && iov != NULL) {
+    delete[] bparr; 
```

Comment:
If you initialize bparr to null where it's declared, you can avoid the need for a conditional here